app/src/main/java/com/android/solvit/seeker/ui/service/Services.kt

```diff
@@ -71,8 +73,16 @@ fun ServicesScreen(
     navigationActions: NavigationActions,
     listProviderViewModel: ListProviderViewModel
 ) {
+
   // Lock Orientation to Portrait
   val localContext = LocalContext.current
+  LaunchedEffect(navigationActions.currentRoute()) {
+    // Clear the selected service when this screen is entered
+    if (navigationActions.currentRoute() == Route.SERVICES) {
+      listProviderViewModel.clearSelectedService()
```

Comment:
Without using LaunchedEffect, the clearSelectedService method would be called every time the composable recomposes. Since we also set the selected service in the composable by removing the launched effect the selected service will never be set ! Do you have an idea on how I can implement it better ?

Reply:
I meant the if condition, the `LaunchedEffect` is fine!